pylint/extensions/check_mccabe.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+"""Module to add McCabe checker class for pylint. """
+
+import ast
+
+from mccabe import McCabeChecker
+from pylint.checkers.base import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import HIGH, IAstroidChecker
+
+
+class MethodMcCabeChecker(BaseChecker):
+    """Checks McCabe complexity cyclomatic threshold in methods and functions
+    to validate a too complex code.
+    """
+
+    __implements__ = IAstroidChecker
+    name = 'methodmccabechecker'
+
+    msgs = {
+        'R1260': (
+            '%s is too complex. The McCabe rating is %d',
+            'too-complex',
+            'Used when a method or function is too complex based on '
+            'McCabe Complexity Cyclomatic'),
+    }
+    options = (
+        ('max-complexity', {
+            'group': 'design',
+            'default': 10,
+            'type': 'int',
+            'metavar': '<int>',
+            'help': 'McCabe complexity cyclomatic threshold',
+        }),
+    )
+
+    @staticmethod
+    def _get_ast_tree(node):
+        """Get a compile python tree of nodes from code"""
+        # TODO: Check if astroid get build a similar tree
+        code = node.as_string()
+        tree = compile(code, '', "exec", ast.PyCF_ONLY_AST)
```

Comment:
@PCManticore 
Could you help to locate a similar code with astroid or pylint (wihtout compile)?
The `type(tree)` is `<class '_ast.Module'>` and the `code` is `def my_method(): pass`.
I did not found a way to use astroid.